opensrp-chw-core/src/main/java/org/smartregister/chw/core/utils/CoreJsonFormUtils.java

```diff
@@ -469,6 +473,12 @@ private static void updateOptions(CommonPersonObjectClient client, JSONObject js
                         Timber.d(e.toString());
                     }
                 }
+
+                if (myKey.equalsIgnoreCase(RECEIVED_DEATH_CERTIFICATE)) fields.put(registrationFormParams.getRight().get(x));
+                if (myKey.equalsIgnoreCase(DEATH_CERTIFICATE_ISSUE_DATE)) fields.put(registrationFormParams.getRight().get(x));
+                if (myKey.equalsIgnoreCase(DEATH_CERTIFICATE_NUMBER)) fields.put(registrationFormParams.getRight().get(x));
+                if (myKey.equalsIgnoreCase(DEATH_NOTIFICATION_DONE)) fields.put(registrationFormParams.getRight().get(x));
```

Comment:
Kindly check the constant against the variable to avoid scenarios where mkey is null.

Reply:
mkey is always non-null